test/test_bot.py

```diff
@@ -366,6 +366,29 @@ def url_handler(*args, **kwargs):
     # no exception implies success
 
 
+# Remove once manual callback management is gone (8.0)
+def test_register_url_callback_manual_warning(tmpconfig, caplog):
+    """Test that manually registering a callback produces a deprecation warning"""
+    test_pattern = r'https://(www\.)?example\.com'
+
+    def url_handler(*args, **kwargs):
+        return None
+
+    sopel = bot.Sopel(tmpconfig, daemon=False)
+    sopel.memory["url_callbacks"] = SopelMemory()
+
+    # register a callback manually
+    sopel.memory["url_callbacks"][re.compile(test_pattern)] = url_handler
+    results = list(sopel.search_url_callbacks("https://www.example.com"))
+    assert results[0][0] == url_handler, "Callback must be present"
+
+    for record in caplog.records:
+        if "url_callbacks" in record.text and "deprecated" in record.text:
```

Comment:
This should (I didn't try it) fix the test on Python != 3.3:

```suggestion
        if "url_callbacks" in record.message and "deprecated" in record.message:
```